### PR TITLE
Fix API methods for working with oauth2 token

### PR DESCRIPTION
**Pull request by sleshchenko**
Add method for retrieving OAuth token
Fix NPE in OAuthAuthenticator
Add RolesAllowed annotation to invalidate method
Reformat the code

### CODE REVIEW

File: platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java [hunk 1]
```diff
@@ -147,13 +152,56 @@ public Response callback() throws OAuthAuthenticationException {
...
+    @GET
+    @Path("/token/invalidate")
+    @RolesAllowed({"user", "temp_user"})
+    public Response invalidate(@Required @QueryParam("oauth_provider") String oauthProvider) throws IOException, BadRequestException {
+        OAuthAuthenticator oauth = getAuthenticator(oauthProvider);
+        final Principal principal = security.getUserPrincipal();
+        if (principal != null && oauth.invalidateToken(principal.getName())) {
+            return Response.ok().build();
+        }
+        return Response.status(404)
```
Comment (garagatyi) [inline platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java]: Why not throw NotFoundException here and set method return type to void

**Comment by garagatyi** [comment]
LGTM

File: platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java (hunk 1 above)
Comment (sleshchenko) [inline platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java]: Nice remark. Thanks

**Comment by skabashnyuk** [comment]
LGTM

**Pull request merged**